version: "3.8"
services:
  # postgres-sol:
  #   image: postgres:latest
  #   container_name: postgres-sol
  #   hostname: postgres-sol
  #   environment:
  #     - POSTGRES_USER=loco
  #     - POSTGRES_PASSWORD=loco
  #     - POSTGRES_DB=sons-of-liberty_development
  #   ports:
  #     - "5432:5432"
  #   volumes:
  #     - postgres-data:/var/lib/postgresql/data
  #   networks:
  #     - sol-network
  #   healthcheck:
  #     test: ["CMD-SHELL", "pg_isready -U loco -d sons-of-liberty_development"]
  #     interval: 5s
  #     timeout: 5s
  #     retries: 5

  sol:
    container_name: sol
    hostname: sol
    build:
      context: ..
      dockerfile: docker/dockerfile
    ports:
      - "5150:5150"
    volumes:
      - ../config:/app/config
    environment:
      - RUST_LOG=info
      - DATABASE_URL=postgres://dlcdevkit:dlcdevkit@host.docker.internal:5433/sons-of-liberty_development
    # depends_on:
    #   postgres-sol:
    #     condition: service_healthy
    extra_hosts:
      - "host.docker.internal:host-gateway"
    networks:
      - sol-network

  sol-frontend:
    container_name: sol-frontend
    hostname: sol-frontend
    build:
      context: ..
      dockerfile: docker/dockerfile-frontend
    ports:
      - "80:80"
    depends_on:
      - sol
    networks:
      - sol-network

volumes:
  postgres-data:

networks:
  sol-network:
    driver: bridge
